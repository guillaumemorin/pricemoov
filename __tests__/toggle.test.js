import React from 'react';
import Toggle from '../src/components/toggle';
import renderer from 'react-test-renderer';

it('Standard output', () => {
    const tree = renderer.create(<Toggle />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Disabled status', () => {
    const tree = renderer.create(<Toggle disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Checked status', () => {
    const tree = renderer.create(<Toggle checked={true} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Display right label', () => {
    const tree = renderer.create(<Toggle labelRight={'Label'} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Display left label', () => {
    const tree = renderer.create(<Toggle labelLeft={'Label'} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Display left label', () => {
    const tree = renderer.create(<Toggle labelLeft={'Label'} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('Display custom background colors', () => {
    const tree = renderer
        .create(
            <Toggle
                backgroundColorChecked={'papayawhip'}
                backgroundColorUnchecked={'aquamarine'}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Set input name and value', () => {
    const tree = renderer
        .create(<Toggle name={'toggle'} value={'toggleValue'} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('Display custom sizes', () => {
    const tree = renderer
        .create(
            <Toggle
                width={40}
                height={20}
                sliderWidth={12}
                sliderHeight={12}
                translate={19}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
