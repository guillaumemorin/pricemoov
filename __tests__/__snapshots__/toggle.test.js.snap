// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checked status 1`] = `
<label
  className="sc-bdVaJa gEhoXm"
>
  <label
    className="sc-bwzfXH kxstlt"
    height={34}
    onChange={undefined}
    width={60}
  >
    <input
      checked={true}
      className="sc-htpNat fiJnNt"
      disabled={undefined}
      name={undefined}
      type="checkbox"
      value={undefined}
    />
    <span
      className="sc-bxivhb bVxZQT"
    />
  </label>
</label>
`;

exports[`Disabled status 1`] = `
<label
  className="sc-bdVaJa gEhoXm"
>
  <label
    className="sc-bwzfXH kxstlt"
    height={34}
    onChange={undefined}
    width={60}
  >
    <input
      checked={undefined}
      className="sc-htpNat fiJnNt"
      disabled={true}
      name={undefined}
      type="checkbox"
      value={undefined}
    />
    <span
      className="sc-bxivhb bVxZQT"
    />
  </label>
</label>
`;

exports[`Display custom background colors 1`] = `
<label
  className="sc-bdVaJa gEhoXm"
>
  <label
    className="sc-bwzfXH kxstlt"
    height={34}
    onChange={undefined}
    width={60}
  >
    <input
      checked={undefined}
      className="sc-htpNat ieUyQK"
      disabled={undefined}
      name={undefined}
      type="checkbox"
      value={undefined}
    />
    <span
      className="sc-bxivhb iFLvUd"
    />
  </label>
</label>
`;

exports[`Display custom sizes 1`] = `
<label
  className="sc-bdVaJa gEhoXm"
>
  <label
    className="sc-bwzfXH kzxgge"
    height={20}
    onChange={undefined}
    width={40}
  >
    <input
      checked={undefined}
      className="sc-htpNat XYBVe"
      disabled={undefined}
      name={undefined}
      type="checkbox"
      value={undefined}
    />
    <span
      className="sc-bxivhb isUKdx"
    />
  </label>
</label>
`;

exports[`Display left label 1`] = `
<label
  className="sc-bdVaJa gEhoXm"
>
  <span
    className="sc-ifAKCX lnyuBs"
  >
    Label
  </span>
  <label
    className="sc-bwzfXH kxstlt"
    height={34}
    onChange={undefined}
    width={60}
  >
    <input
      checked={undefined}
      className="sc-htpNat fiJnNt"
      disabled={undefined}
      name={undefined}
      type="checkbox"
      value={undefined}
    />
    <span
      className="sc-bxivhb bVxZQT"
    />
  </label>
</label>
`;

exports[`Display left label 2`] = `
<label
  className="sc-bdVaJa gEhoXm"
>
  <span
    className="sc-ifAKCX lnyuBs"
  >
    Label
  </span>
  <label
    className="sc-bwzfXH kxstlt"
    height={34}
    onChange={undefined}
    width={60}
  >
    <input
      checked={undefined}
      className="sc-htpNat fiJnNt"
      disabled={undefined}
      name={undefined}
      type="checkbox"
      value={undefined}
    />
    <span
      className="sc-bxivhb bVxZQT"
    />
  </label>
</label>
`;

exports[`Display right label 1`] = `
<label
  className="sc-bdVaJa gEhoXm"
>
  <label
    className="sc-bwzfXH kxstlt"
    height={34}
    onChange={undefined}
    width={60}
  >
    <input
      checked={undefined}
      className="sc-htpNat fiJnNt"
      disabled={undefined}
      name={undefined}
      type="checkbox"
      value={undefined}
    />
    <span
      className="sc-bxivhb bVxZQT"
    />
  </label>
  <span
    className="sc-ifAKCX lnyuBs"
  >
    Label
  </span>
</label>
`;

exports[`Set input name and value 1`] = `
<label
  className="sc-bdVaJa gEhoXm"
>
  <label
    className="sc-bwzfXH kxstlt"
    height={34}
    onChange={undefined}
    width={60}
  >
    <input
      checked={undefined}
      className="sc-htpNat fiJnNt"
      disabled={undefined}
      name="toggle"
      type="checkbox"
      value="toggleValue"
    />
    <span
      className="sc-bxivhb bVxZQT"
    />
  </label>
</label>
`;

exports[`Standard output 1`] = `
<label
  className="sc-bdVaJa gEhoXm"
>
  <label
    className="sc-bwzfXH kxstlt"
    height={34}
    onChange={undefined}
    width={60}
  >
    <input
      checked={undefined}
      className="sc-htpNat fiJnNt"
      disabled={undefined}
      name={undefined}
      type="checkbox"
      value={undefined}
    />
    <span
      className="sc-bxivhb bVxZQT"
    />
  </label>
</label>
`;
